### PR TITLE
GH-125 expand default config and PlaceholderAPI support in internal placeholders

### PR DESCRIPTION
**Pull request by Kamicjusz**
This PR add support for PlaceholderAPI in intenal placeholders
also expand default plugin configuration

### CODE REVIEW

**Review by P1otrulla — APPROVED** [review]

File: chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java [hunk 1]
```diff
@@ -44,31 +47,40 @@ public class PluginConfig implements ChatSettings, PlaceholderStack, TemplateRep
         "# RGB and gradient message usage: ",
         "# <color:#ff00ee>Example message</color>",
         "# <gradient:#ff00ee:#f79459>Example message</gradient>",
-        " "
+        "# ",
+        "# You can use three more placeholders: <displayname> <name> <message>",
```
Comment (CitralFlo) [inline chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:51]: The plugin includes placeholders: 
<displayname> - in game nick
<name> -  ??? (please describle)
<message> - message sent by the player

File: chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java [hunk 2]
```diff
@@ -25,7 +28,7 @@ public class PluginConfig implements ChatSettings, PlaceholderStack, TemplateRep
     public boolean receiveUpdates = true;
 
 
-    @Description({ " ", "# Chat format for ranks (Vault) Support mini-messages and legacy colors" })
+    @Description({ " ", "# Chat rankFormat for ranks (Vault) Support mini-messages and legacy colors" })
     @Description({ " ", "# We're recommending to use webui for mini-messages: https://webui.adventure.kyori.net/" })
     @Description({ " ", "# documentation is here: https://docs.adventure.kyori.net/minimessage/format.html" })
```
Comment (CitralFlo) [inline chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:34]: `@Description({ " ", "# You can check LuckyPerms setup and placeholders here: https://luckperms.net/wiki/Placeholders" })`

File: chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java [hunk 3]
```diff
@@ -25,7 +28,7 @@ public class PluginConfig implements ChatSettings, PlaceholderStack, TemplateRep
     public boolean receiveUpdates = true;
 
 
-    @Description({ " ", "# Chat format for ranks (Vault) Support mini-messages and legacy colors" })
+    @Description({ " ", "# Chat rankFormat for ranks (Vault) Support mini-messages and legacy colors" })
```
Comment (vLuckyyy) [inline chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:31]: Can you stay at the `format`? That makes breaking change. This change can be break old user's config.

File: chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java (hunk 1 above)
Comment (vLuckyyy) [reply chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:51]: Name, displayname, message

Comment (vLuckyyy) [reply chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:51]: Displayname, name, message

Comment (CitralFlo) [reply chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:51]: I mean, polish people will use our plugin and You know they need **special** explanation

File: chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java [hunk 4]
```diff
@@ -28,8 +31,10 @@ public class PluginConfig implements ChatSettings, PlaceholderStack, TemplateRep
     @Description({ " ", "# Chat format for ranks (Vault) Support mini-messages and legacy colors" })
     @Description({ " ", "# We're recommending to use webui for mini-messages: https://webui.adventure.kyori.net/" })
     @Description({ " ", "# documentation is here: https://docs.adventure.kyori.net/minimessage/format.html" })
+    @Description({ " ", "# You can check LuckyPerms setup and placeholders here: https://luckperms.net/wiki/Placeholders" })
```
Comment (igoyek) [inline chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:34]: LuckPerms ;)

File: chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java [hunk 5]
```diff
@@ -44,31 +49,43 @@ public class PluginConfig implements ChatSettings, PlaceholderStack, TemplateRep
...
         "# <gradient:#ff00ee:#f79459>Example message</gradient>",
-        " "
+        "# ",
+        "# You can use three more internal placeholders: <displayname> <name> <message>",
+        "# "
     })
-    public String defaultFormat = "{displayname} {arrow_right} {message}";
+    public String defaultFormat = "{displayname} » {message}";
+    @Description({ " ", "# Here you can set different formats for each rank.",
+            "# Remember! Rank name must be exact same like in you permission plugin configuration.",
```
Comment (CitralFlo) [inline chatformatter-core/src/main/java/com/eternalcode/formatter/config/PluginConfig.java:58]: Rank name must be **exactly the same as in** your permission plugin configuration.

**Review by CitralFlo — APPROVED** [review]
Minor lang error

File: build.gradle.kts [hunk 6]
```diff
@@ -85,7 +85,7 @@ runPaper {
 }
 
 tasks.runServer {
-    minecraftVersion("1.20.1")
+    minecraftVersion("1.20.4")
```
[the remark this answers is not in the document]
Comment (vLuckyyy) [reply build.gradle.kts:88]: In 1.20.6, Mojang Studio said they'll switch to Java 21. If we switch to 1.20.6, we won't be able to start the server without changing Java to 21 for Gradle. I'm just saying this in a roundabout way (I assume you've already learned from @Kamicjusz).

**Review by vLuckyyy — APPROVED** [review]
These changes are very good, I also like your cooperation. Keep it up guys!

**Review by Jakubk15 — APPROVED** [review]
sweet

**Pull request merged**